docs/services/ml/upload-training-script.md

```diff
@@ -10,48 +10,231 @@ no_service: true
 ---
 
 You can create your own custom Python training script that trains ML models to your specifications using the Machine Learning framework of your choice (PyTorch, Tensorflow, TFLite, ONNX, or any other framework).
-Once added to the [Viam registry](https://app.viam.com/registry?type=Training+Script), you can use the training script to build models for your datasets.
+Once added to the [Viam registry](https://app.viam.com/registry?type=Training+Script), you can use the training script to build ML models based on your datasets.
 
 {{< alert title="Note" color="note" >}}
-For a lower-code approach, you can [train TFLite classification and object detection models](/services/ml/train-model/) on the [Viam app **DATA** page](https://app.viam.com).
+For a no-code approach, you can [train TFLite classification and object detection models](/services/ml/train-model/) on the [Viam app **DATA** page](https://app.viam.com) or use existing [training scripts](https://app.viam.com/registry?type=Training+Script) from the Viam Registry.
 {{< /alert >}}
 
 Follow this guide to create, upload, and submit a Python script that loads a training dataset, trains an ML model, and produces a model artifact.
 
 ## Create a training script
 
-To start, create an entrypoint file called <file>training.py</file> in a folder called <file>model</file>.
-This file will contain the main logic of your training script.
-
-Add the following logic to your code in <file>training.py</file>:
-
-{{%expand "Step 1: Parse command line inputs" %}}
+1. Create a folder for the training-script, for example <file>my-training</file>.
+1. Inside the top level folder (in this example <file>my-training</file>), create a file called `setup.py` with the following contents:
+
+   ```python {class="line-numbers linkable-line-numbers" data-line="11"}
+   from setuptools import find_packages, setup
+
+   setup(
+       name="my-training",
+       version="0.1",
+       packages=find_packages(),
+       include_package_data=True,
+       install_requires=[
+           "google-cloud-aiplatform",
+           "google-cloud-storage",
+           # TODO: Add additional required packages
+       ],
+   )
+   ```
+
+   Ensure you add additional required packages on line 11.
+
+1. Inside the top level folder (in this example <file>my-training</file>), create a folder called <file>model</file> and create an empty file called <file>\_\_init\_\_.py</file> inside it.
+1. Inside the <file>model</file> folder, create a file called <file>training.py</file>.
+
+   {{< alert title="Using Viam APIs in a training script" color="note" >}}
+   If you need to access any of the [Viam APIs](/appendix/apis/) within a custom training script, you can use the environment variables `API_KEY` and `API_KEY_ID` to establish a connection.
```

Comment:
Do you think it's worth having a snippet for the connection?

```
async def connect() -> ViamClient:
    """Returns a authenticated connection to the ViamClient for the requested org associated with the submitted training job."""
    # The API key and key ID can be accessed programmatically, using the environment variable API_KEY and API_KEY_ID.
    # The user does not need to supply the API keys, they are provided automatically when the training job is submitted.
    dial_options = DialOptions.with_api_key(
        os.environ.get("API_KEY"), os.environ.get("API_KEY_ID")
    )
    return await ViamClient.create_from_dial_options(dial_options)
```

Reply:
We could also just say where to access this code snippet in app